kong/plugins/acl/handler.lua

```diff
@@ -99,10 +104,20 @@ function ACLHandler:access(conf)
       end
 
       if not consumer_groups then
-        local status = kong.client.get_credential() and 403 or 401
-        return kong.response.exit(status, {
-          message = "You cannot consume this service"
-        })
+        if config.type == BLACK and kong.client.get_credential() then
+          consumer_groups = EMPTY
+
+        else
+          if kong.client.get_credential() then
```

Comment:
This code calls `kong.client.get_credential()` twice. LuaJIT will probably optimize this up, but let's localize it just in case.